[Canvas] Feat: Zoom In/Out

cqliu1 opened the pull request:
## Summary

Closes #27123.
This adds the ability to zoom in/out of the work space in Canvas.

![Jun-13-2019 13-23-00](https://user-images.githubusercontent.com/1697105/59464981-b165b680-8dde-11e9-9f44-b874b8a0d28f.gif)

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1773/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1961/)

w33ble commented on x-pack/plugins/canvas/public/state/initial_state.js:
```diff
@@ -14,6 +14,7 @@ export const getInitialState = path => {
     assets: {}, // assets end up here
     transient: {
       canUserWrite: capabilities.get().canvas.save,
+      zoomLevel: 1,
```
`zoomScale`?

w33ble commented on x-pack/plugins/canvas/common/lib/constants.ts:
```diff
@@ -25,3 +25,6 @@ export const DEFAULT_WORKPAD_CSS = '.canvasPage {\n\n}';
 export const DEFAULT_ELEMENT_CSS = '.canvasRenderEl{\n\n}';
 export const VALID_IMAGE_TYPES = ['gif', 'jpeg', 'png', 'svg+xml'];
 export const ASSET_MAX_SIZE = 25000;
+export const ZOOM_LEVELS = [0.5, 0.75, 0.9, 1, 1.5, 2, 3, 4];
```
`[0.2, 0.25, 0.5, 0.75, 1, 1.5, 2, 3, 4]` Allows 5K resolution on 1K screens, FWIW.

ryankeairns commented:
@cqliu1 design PR to delineate top and side sections, so that the page will appear to go under them as you zoom in or scroll around: :octocat:  https://github.com/cqliu1/kibana/pull/9

monfera commented:
retest

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2036/)

andreadelrio commented:
![canvas - zoom - openMenu](https://user-images.githubusercontent.com/4016496/59530487-8b045180-8e98-11e9-884d-0b2d1a2a27b4.png)

@cqliu1 what do you think of this?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2376/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2411/)

cqliu1 commented:
> ![canvas - zoom - openMenu](https://user-images.githubusercontent.com/4016496/59530487-8b045180-8e98-11e9-884d-0b2d1a2a27b4.png)
> 
> @cqliu1 what do you think of this?

It looks good! I think I'm going to leave the keyboard shortcuts off the menu item for now though. I've talked with @ryankeairns about adding the shortcuts, but I don't think it's possible to add them yet.

cqliu1 commented:
@andreadelrio
![Screen Shot 2019-06-17 at 2 44 44 PM](https://user-images.githubusercontent.com/1697105/59654725-5f10f680-914c-11e9-911f-fdd12f94a365.png)
 I added the context menu for the zoom actions. I added a divider border per @ryankeairns request.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2494/)

ryankeairns commented:
@cqliu1 the zoom icons have been merged on the EUI side, so we'll just have to wait for the next EUI build and subsequent Kibana update.

In the meantime, we could try out the icons if you'd like to store the SVGs in Kibana (`EuiIcon` now accepts a ref to an SVG file) and later remove them once master is updated. Here are the files:

[ 🗄:arrow_down: zoom-icons.zip](https://github.com/elastic/kibana/files/3306609/zoom-icons.zip)

#### Preview
<img width="483" alt="Screenshot 2019-06-19 08 36 45" src="https://user-images.githubusercontent.com/446285/59775691-346f8c80-9277-11e9-9328-711d40c4a536.png">

elasticmachine commented:
Pinging @elastic/kibana-canvas

cqliu1 commented:
@ryankeairns I'm okay with waiting for the next EUI update as long as it gets into to 7.3.0 before FF.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2916/)

ryankeairns commented:
The plan is to start a new EUI release tomorrow (6/21)... stay tuned 📺

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3835/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3855/)

ambigus9 commented:
Excelent!!!

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4490/)
